ext/pdo_pgsql/pgsql_driver.c

```diff
@@ -361,31 +362,37 @@ static char *pdo_pgsql_last_insert_id(pdo_dbh_t *dbh, const char *name, size_t *
 {
 	pdo_pgsql_db_handle *H = (pdo_pgsql_db_handle *)dbh->driver_data;
 	char *id = NULL;
+	char *version = NULL;
 
-	if (name == NULL) {
-		if (H->pgoid == InvalidOid) {
-			return NULL;
-		}
-		*len = spprintf(&id, 0, ZEND_LONG_FMT, (zend_long) H->pgoid);
+	PGresult *res;
+	ExecStatusType status;
+	const char *q[1];
+	q[0] = name;
+
+	res = PQexec(H->server, "SHOW server_version_num");
+	version = estrdup((char *)PQgetvalue(res, 0, 0));
```

Comment:
Looking at libpq it looks like there is a `PQserverVersion` function that directly returns the version as an integer.

However I'm not sure if there's some kind of problem with this function as it is not used here either: https://github.com/phackwer/php-src/blob/58ea58609ea292273ba8c8e91a52c8a04c82ee01/ext/pdo_pgsql/pgsql_driver.c#L416